rayzardemo.py

```diff
@@ -42,37 +38,48 @@ def load(location, filter=None):
     p = subprocess.run(args=args, stdout=subprocess.PIPE)
     return p.stdout
 
+# ArchiveSearch implements a ray actor to perform a zar search and provide
+# a stream of hits via the next() method.
 @ray.remote
-class Archive(object):
-    def __init__(self, root):
-        self.root = root
-
-    def find(self, findq):
-        args = [zarexec, 'find', '-R', self.root, findq]
+class ArchiveSearch(object):
+    def __init__(self, root, findq):
+        # we run the find all at once up front... it would be better to stream
+        # this if this weren't a demo
+        args = [zarexec, 'find', '-R', root, findq]
         p = subprocess.run(args=args, stdout=subprocess.PIPE, text=True)
-        locs = p.stdout.splitlines()
-        return [load.remote(loc) for loc in locs]
-
-@ray.remote
-def zqcombine(zql, obj_ids):
-    zng = bytearray()
-    for o in obj_ids:
-        zng.extend(ray.get(o))
-    return localzq(zql, zng)
-
+        self.locs = p.stdout.splitlines()
+
+    @ray.method(num_return_vals=1)
```

Comment:
I believe the `ray.method` annotation is only necessary if you need to specify that the number of object ids returned by the method will be something other than 1.

Reply:
That is correct.  Will fix.